provider/rfc2136/rfc2136.go

```diff
@@ -439,3 +450,18 @@ func chunkBy(slice []*endpoint.Endpoint, chunkSize int) [][]*endpoint.Endpoint {
 
 	return chunks
 }
+
+func findMsgZone(ep *endpoint.Endpoint, m *dns.Msg, r *rfc2136Provider) {
```

Comment:
please return string and set/update the msg and provider members outside this func.

Reply:
fixed: https://github.com/kubernetes-sigs/external-dns/pull/3976/commits/7fb529965ddef54bf5489b4bc7984d20409ba3a6